### ISSUE TITLE
Error en escritura del rut con modo live en safari

### ISSUE THREAD

**Issue by richardgarces** [platanus/angular-rut#9]
Hay un error en safari, en modo live, al escribir el rut, no queda bien escrito.

**Comment by bunzli** [comment]
@richardgarces no logro reproducir el problema con Safari. Nos puedes dar más información?, cuál es el rut que estás escribiendo?

**Comment by richardgarces** [comment]
Con cualquier RUT, por ejemplo 191 queda escrito como 11-9

**Comment by bunzli** [comment]
en serio?, da vuelta el 1 por el 9? Lo correcto, para ese caso, sería mostrar 19-1 cierto?

**Comment by richardgarces** [comment]
Exacto y con rut más grande, da vuelta más números, solo en safari.

**Comment by richardgarces** [comment]
y en modo blur, deja ingresar letras, caracteres no válidos para un rut.

El 23 de marzo de 2015, 12:24, Richard Garcés Riffo <richardgarces@gmail.com
> escribió:

> Exacto y con rut más grande, da vuelta más números, solo en safari.
>
>
>


-- 
Richard A. Garcés Riffo
http://www.flickr.com/rgarces


"Quien tiene un porqué para vivir podrá superar cualquier cómo" (Friedrich
Nietzsche)

**Comment by cassandrainside** [comment]
Y podrias decirme "como" desactivo las threads

**Comment by agustinf** [comment]
@richardgarces en qué versión de Safari te da problemas?

**Comment by richardgarces** [comment]
no entiendo tu consulta.

2015-03-26 11:13 GMT-03:00 cassandrainside <notifications@github.com>:

> Y podrias decirme "como" desactivo las threads
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/platanus/angular-rut/issues/9#issuecomment-86531826>.
>



-- 
Richard A. Garcés Riffo
http://www.flickr.com/rgarces


"Quien tiene un porqué para vivir podrá superar cualquier cómo" (Friedrich
Nietzsche)

**Comment by richardgarces** [comment]
Versión: 5.1.7 (7534.57.2)

2015-03-26 11:29 GMT-03:00 Agustin Feuerhake <notifications@github.com>:

> @richardgarces <https://github.com/richardgarces> en qué versión de
> Safari te da problemas?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/platanus/angular-rut/issues/9#issuecomment-86541013>.
>



-- 
Richard A. Garcés Riffo
http://www.flickr.com/rgarces


"Quien tiene un porqué para vivir podrá superar cualquier cómo" (Friedrich
Nietzsche)

**Comment by agustinf** [comment]
Y usas Safari en Windows o Mac?

**Comment by richardgarces** [comment]
Sucede en Windows y en Mac

2015-03-26 11:44 GMT-03:00 Agustin Feuerhake <notifications@github.com>:

> Y usas Safari en Windows o Mac?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/platanus/angular-rut/issues/9#issuecomment-86545113>.
>



-- 
Richard A. Garcés Riffo
http://www.flickr.com/rgarces


"Quien tiene un porqué para vivir podrá superar cualquier cómo" (Friedrich
Nietzsche)

**Comment by agustinf** [comment]
Efectivamente. Lo pude replicar usando un Safari 5.1 en www.browserstack.com.

**Comment by richardgarces** [comment]
Que bien,ya estaba dudando... jajajja

2015-03-26 12:09 GMT-03:00 Agustin Feuerhake <notifications@github.com>:

> Efectivamente. Lo pude replicar usando un Safari 5.1 en
> www.browserstack.com.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/platanus/angular-rut/issues/9#issuecomment-86560863>.
>



-- 
Richard A. Garcés Riffo
http://www.flickr.com/rgarces


"Quien tiene un porqué para vivir podrá superar cualquier cómo" (Friedrich
Nietzsche)

**Comment by agustinf** [comment]
Una idea puede ser usar algo como esto para asegurarnos de que el cursor quede al final cuando la persona hace el key up http://jsfiddle.net/ghAB9/3/ pero hay que tener cuidado de no interrumpir si es que está usando las flechas para mover el cursor...

**Comment by cassandrainside** [comment]
Muchas gracia. Podría decirme para cuando sería el magno evento, todos los cientificos dicen que a más tardar mediados del 16? ha leido el libro Ramatis, mensajes del astral?

**Comment by bunzli** [comment]
creo que no basta con mandarlo al final. De hecho creo que hay que dar soporte de mantener el cursor en el lugar correcto luego del format live... en chrome siempre se va al final, y es bastante molesto si se quiere editar algo entre medio.

**Comment by bunzli** [comment]
El format live tiene un par de dificultades que debemos resolver.

¿queremos poder editar un rut "entre medio" con format live?, En este caso, ¿qué pasa si borramos un "." o un "-"?, el format lo vuelve a agregar.... no se me ocurre que debiera pasar.

Tiendo a pensar que en format live, el cursor siempre debiéramos forzar el cursor al final del texto, aunque haga click "entre medio". Si quiero editar, tengo que borrar hasta llegar al error.

¿qué opinan? @agustinf @blackjid @ldlsegovia @ReneMoraales

**Comment by richardgarces** [comment]
Opino que el formato live, si elimina debiera eliminar del último caracter
hacia atrás, ya que coloca los puntos y guión automáticamente.

El 27 de marzo de 2015, 11:27, Jaime Bunzli <notifications@github.com>
escribió:

> El format live tiene un par de dificultades que debemos resolver.
>
> ¿queremos poder editar un rut "entre medio" con format live?, En este
> caso, ¿qué pasa si borramos un "." o un "-"?, el format lo vuelve a
> agregar.... no se me ocurre que debiera pasar.
>
> Tiendo a pensar que en format live, el cursor siempre debiéramos forzar el
> cursor al final del texto, aunque haga click "entre medio". Si quiero
> editar, tengo que borrar hasta llegar al error.
>
> ¿qué opinan? @agustinf <https://github.com/agustinf> @blackjid
> <https://github.com/blackjid> @ldlsegovia <https://github.com/ldlsegovia>
> @ReneMoraales <https://github.com/ReneMoraales>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/platanus/angular-rut/issues/9#issuecomment-86957397>.
>



-- 
Richard A. Garcés Riffo
http://www.flickr.com/rgarces


"Quien tiene un porqué para vivir podrá superar cualquier cómo" (Friedrich
Nietzsche)

**Comment by bunzli** [comment]
entonces @richardgarces tú eres de la opinión de forzar siempre el cursor al final?

**Comment by richardgarces** [comment]
para el caso de live

El 27 de marzo de 2015, 11:36, Richard Garcés Riffo <richardgarces@gmail.com
> escribió:

> Para ese caso sí.
>
> 2015-03-27 11:33 GMT-03:00 Jaime Bunzli <notifications@github.com>:
>
>> entonces @richardgarces <https://github.com/richardgarces> tú eres de la
>> opinión de forzar siempre el cursor al final?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/platanus/angular-rut/issues/9#issuecomment-86959287>.
>>
>
>
>
> --
> Richard A. Garcés Riffo
> http://www.flickr.com/rgarces
>
>
> "Quien tiene un porqué para vivir podrá superar cualquier cómo" (Friedrich
> Nietzsche)
>



-- 
Richard A. Garcés Riffo
http://www.flickr.com/rgarces


"Quien tiene un porqué para vivir podrá superar cualquier cómo" (Friedrich
Nietzsche)

**Comment by richardgarces** [comment]
Para ese caso sí.

2015-03-27 11:33 GMT-03:00 Jaime Bunzli <notifications@github.com>:

> entonces @richardgarces <https://github.com/richardgarces> tú eres de la
> opinión de forzar siempre el cursor al final?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/platanus/angular-rut/issues/9#issuecomment-86959287>.
>



-- 
Richard A. Garcés Riffo
http://www.flickr.com/rgarces


"Quien tiene un porqué para vivir podrá superar cualquier cómo" (Friedrich
Nietzsche)

**Comment by bunzli** [comment]
sí sí...  para el live solamente

**Comment by ReneMoraales** [comment]
sí, estoy de acuerdo. creo que cualquiera sea la decisión que tomásemos respecto a la "edición en el medio", dudo que sea 100% claro el comportamiento para el usuario.

**Comment by blackjid** [comment]
no se si soluciona exactamente el problema, pero... tal vez se podria poner un threshold de cantidad de caracteres para el live format. Por ejemplo que no formate hasta que se escriban por lo menos 7 caracteres.. que seria un rut de un viejito tipo `243.234-8`  encuentro muy raro el live cuando vas en dos caracteres y tiene `2-4`

**Comment by RikRed** [comment]
Esto tambien fue detectado en el proyecto en que trabajo. Saben para cuando estaria la solucion? Safari es el unico browser que no ha pasado los casos de prueba de rut en QA.... :S

**Comment by bunzli** [comment]
No estamos realmente seguros de como queremos dejar la funcionalidad de formateo "live". Nos va a tomar un poco más de tiempo decidir esto. 

@RikRed el format-blur funciona bien en safari, cierto?

**Comment by richardgarces** [comment]
blur, permite ingresar cualquier caracter, live en otros browser si cambio
un número en medio el cursor se va al final, creo que sería el mismo
comportamiento para live en safari.

El 2 de abril de 2015, 12:30, Jaime Bunzli <notifications@github.com>
escribió:

> No estamos realmente seguros de como queremos dejar la funcionalidad de
> formateo "live". Nos va a tomar un poco más de tiempo decidir esto.
>
> @RikRed <https://github.com/RikRed> el format-blur funciona bien en
> safari, cierto?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/platanus/angular-rut/issues/9#issuecomment-88948118>.
>



-- 
Richard A. Garcés Riffo
http://www.flickr.com/rgarces


"Quien tiene un porqué para vivir podrá superar cualquier cómo" (Friedrich
Nietzsche)

**Comment by RikRed** [comment]
al modo blur solo le falta aplicar alguna directiva para que solo permita numeros y la letra K (k).

**Comment by bunzli** [comment]
estarían de acuerdo con este issue?

https://github.com/platanus/angular-rut/issues/10

**Comment by jaimeguaman** [comment]
Estoy teniendo este problema con Chrome Android

    ```
     Android 6.0.1
     Chrome 56.0.2924.87 ```

 
![screenshot_20170213-020003](https://cloud.githubusercontent.com/assets/1824920/22871578/c8155da0-f190-11e6-927c-0a2abeaf7749.png )


![screenshot_20170213-015948](https://cloud.githubusercontent.com/assets/1824920/22871577/c81311da-f190-11e6-833a-889281b1e2de.png)
